Makefile

```diff
@@ -30,7 +30,7 @@ MLRUN_NO_CACHE ?=
 MLRUN_ML_DOCKER_IMAGE_NAME_PREFIX ?= ml-
 # do not specify the patch version so that we can easily upgrade it when needed - it is determined by the base image
 # mainly used for mlrun, base and mlrun-gpu. mlrun API version >= 1.3.0 should always have python 3.9
-MLRUN_PYTHON_VERSION ?= 3.9
+MLRUN_PYTHON_VERSION ?= 3.11
```

Comment:
I suggest using this Python version in the `uv pip compile` targets for the relevant images - with `mlrun/mlrun-api` as the leading example.
The following unchanged setting in `pyproject.toml` should define the minimal Python version:
https://docs.astral.sh/uv/reference/settings/#pip_python-version
But it can be overridden with this flag:
https://docs.astral.sh/uv/reference/cli/#uv-pip-compile--python-version
https://github.com/mlrun/mlrun/blob/f9dd4303ba335deb5a14df4ff2b442b59d09a100/Makefile#L884-L891

Obviously this comment is irrelevant for images that must support Python 3.9 as well, such as `mlrun/mlrun`.